### ISSUE TITLE
`nix-shell` fails to start on High Sierra

### ISSUE THREAD

**Issue by themattchan** [awakesecurity/proto3-suite#47]
Probably just have to update `nixpkgs` from 16 to 17?

```
$ nix-shell -j8 --cores 0 release.nix -A proto3-suite.env 
these derivations will be built:
  /nix/store/a590rgix4n91szf7pfrs9c0n41ysfvgz-proto3-suite-0.1.0.0.drv
  /nix/store/r158k5srdddsnq3sdmnf5na4d1x86j2w-ghc-8.0.1-with-packages.drv
building path(s) ‘/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0’
setupCompilerEnvironmentPhase
Build with /nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1.
unpacking sources
unpacking source archive /nix/store/9vcncmy77m1ks66pyd4q72mcyin8kp8s-proto3-suite
source root is proto3-suite
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/private/var/folders/fb/845d7wfd21g3flzcchdmcq740000gn/T/nix-build-proto3-suite-0.1.0.0.drv-0/package.conf.d -j8 -threaded
[1 of 1] Compiling Main             ( /nix/store/4mdp8nhyfddh7bllbi7xszz7k9955n79-Setup.hs, /private/var/folders/fb/845d7wfd21g3flzcchdmcq740000gn/T/nix-build-proto3-suite-0.1.0.0.drv-0/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --with-gcc=clang --package-db=/private/var/folders/fb/845d7wfd21g3flzcchdmcq740000gn/T/nix-build-proto3-suite-0.1.0.0.drv-0/package.conf.d --ghc-option=-optl=-Wl,-headerpad_max_install_names --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --enable-library-vanilla --enable-executable-dynamic --disable-tests
Configuring proto3-suite-0.1.0.0...
Dependency QuickCheck >=2.8 && <2.10: using QuickCheck-2.8.2
Dependency aeson ==1.1.1.0: using aeson-1.1.1.0
Dependency aeson-pretty -any: using aeson-pretty-0.8.2
Dependency attoparsec >=0.13.0.1: using attoparsec-0.13.1.0
Dependency base >=4.8 && <5.0: using base-4.9.0.0
Dependency base64-bytestring >=1.0.0.1 && <1.1: using
base64-bytestring-1.0.0.1
Dependency bytestring >=0.10.6.0 && <0.11.0: using bytestring-0.10.8.1
Dependency cereal >=0.5.1 && <0.6: using cereal-0.5.4.0
Dependency containers ==0.5.*: using containers-0.5.7.1
Dependency deepseq ==1.4.*: using deepseq-1.4.2.0
Dependency foldl -any: using foldl-1.2.1
Dependency haskell-src ==1.0.*: using haskell-src-1.0.2.0
Dependency lens -any: using lens-4.14
Dependency mtl ==2.2.*: using mtl-2.2.1
Dependency neat-interpolation -any: using neat-interpolation-0.3.2.1
Dependency optparse-generic -any: using optparse-generic-1.2.1
Dependency parsec >=3.1.9 && <3.2.0: using parsec-3.1.11
Dependency parsers ==0.12.*: using parsers-0.12.4
Dependency pretty ==1.1.*: using pretty-1.1.3.3
Dependency pretty-show >=1.6.12 && <1.7: using pretty-show-1.6.12
Dependency proto3-suite -any: using proto3-suite-0.1.0.0
Dependency proto3-wire ==1.0.*: using proto3-wire-1.0.0
Dependency range-set-list >=0.1.2 && <0.2: using range-set-list-0.1.2.0
Dependency safe ==0.3.*: using safe-0.3.10
Dependency semigroups ==0.18.*: using semigroups-0.18.2
Dependency swagger2 ==2.1.6: using swagger2-2.1.6
Dependency system-filepath -any: using system-filepath-0.4.13.4
Dependency text >=0.2 && <1.3: using text-1.2.2.1
Dependency transformers >=0.4 && <0.6: using transformers-0.5.2.0
Dependency turtle -any: using turtle-1.3.6
Dependency vector >=0.11 && <0.13: using vector-0.11.0.0
Using Cabal-1.24.0.0 compiled by ghc-8.0
Using compiler: ghc-8.0.1
Using install prefix:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0
Binaries installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/bin
Libraries installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/lib/ghc-8.0.1/proto3-suite-0.1.0.0
Private binaries installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/libexec
Data files installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/share/x86_64-osx-ghc-8.0.1/proto3-suite-0.1.0.0
Documentation installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/share/doc/x86_64-osx-ghc-8.0.1/proto3-suite-0.1.0.0
Configuration files installed in:
/nix/store/9q71mbb6aw9h1rigf3mkz0r80wz4br45-proto3-suite-0.1.0.0/etc
No alex found
Using ar found on system at:
/nix/store/wapvwwm07lrgxwi1ahpqy5a2i6k2fw4n-cctools-binutils-darwin/bin/ar
No c2hs found
No cpphs found
Using gcc version 4.2.1 given by user at:
/nix/store/jlrnkff9r2dwspvwr053nki6nns43qfv-clang-wrapper-3.7.1/bin/clang
Using ghc version 8.0.1 found on system at:
/nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1/bin/ghc
Using ghc-pkg version 8.0.1 found on system at:
/nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.17.2 found on system at:
/nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1/bin/hpc
Using hsc2hs version 0.68 found on system at:
/nix/store/63rvbk2f3xcpk0fbcjgjzdbyv6nbkl8k-ghc-8.0.1/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/ifs67km6xh84hcxsssfwa139pr1xm7v7-hscolour-1.24.1/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/jlrnkff9r2dwspvwr053nki6nns43qfv-clang-wrapper-3.7.1/bin/ld
No lhc found
No lhc-pkg found
No pkg-config found
Using strip found on system at:
/nix/store/wapvwwm07lrgxwi1ahpqy5a2i6k2fw4n-cctools-binutils-darwin/bin/strip
Using tar found on system at:
/nix/store/j4y59y75sk1jprcz8drhhamsm286cdwa-gnutar-1.29/bin/tar
No uhc found
building
Building proto3-suite-0.1.0.0...
Preprocessing library proto3-suite-0.1.0.0...
[ 1 of 13] Compiling Proto3.Suite.DotProto.AST ( src/Proto3/Suite/DotProto/AST.hs, dist/build/Proto3/Suite/DotProto/AST.o )
[ 2 of 13] Compiling Proto3.Suite.DotProto.Parsing ( src/Proto3/Suite/DotProto/Parsing.hs, dist/build/Proto3/Suite/DotProto/Parsing.o )

src/Proto3/Suite/DotProto/Parsing.hs:65:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘listSep’
[ 3 of 13] Compiling Proto3.Suite.DotProto.Rendering ( src/Proto3/Suite/DotProto/Rendering.hs, dist/build/Proto3/Suite/DotProto/Rendering.o )
[ 4 of 13] Compiling Proto3.Suite.Types ( src/Proto3/Suite/Types.hs, dist/build/Proto3/Suite/Types.o )
[ 5 of 13] Compiling Proto3.Suite.DotProto ( src/Proto3/Suite/DotProto.hs, dist/build/Proto3/Suite/DotProto.o )
[ 6 of 13] Compiling Proto3.Suite.DotProto.Internal ( src/Proto3/Suite/DotProto/Internal.hs, dist/build/Proto3/Suite/DotProto/Internal.o )
<command line>: can't load .so/.DLL for: /nix/store/65xh2q7v9mgbyqkyn1h7imaqkjyq4szq-libiconv-osx-10.9.5/lib/libiconv.dylib (dlopen(/nix/store/65xh2q7v9mgbyqkyn1h7imaqkjyq4szq-libiconv-osx-10.9.5/lib/libiconv.dylib, 5): no suitable image found.  Did find:
	/nix/store/65xh2q7v9mgbyqkyn1h7imaqkjyq4szq-libiconv-osx-10.9.5/lib/libiconv.dylib: malformed mach-o image: symbol table underruns __LINKEDIT
	/nix/store/65xh2q7v9mgbyqkyn1h7imaqkjyq4szq-libiconv-osx-10.9.5/lib/libiconv.dylib: malformed mach-o image: symbol table underruns __LINKEDIT)
builder for ‘/nix/store/a590rgix4n91szf7pfrs9c0n41ysfvgz-proto3-suite-0.1.0.0.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/r158k5srdddsnq3sdmnf5na4d1x86j2w-ghc-8.0.1-with-packages.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/r158k5srdddsnq3sdmnf5na4d1x86j2w-ghc-8.0.1-with-packages.drv’ failed
/Users/matt/.nix-profile/bin/nix-shell: failed to build all dependencies
```

**Comment by intractable** [comment]
Thanks, I'll take a look ASAP.

**Comment by themattchan** [comment]
Update: 

I updated `nixpkgs.json` to `release-17.09`with the following command:

 ```nix-prefetch-git --out nixpkgs.json  https://github.com/NixOS/nixpkgs.git fd0abdcebc121c8685e3f1e514f022c59a81b2b5```

There is a breaking change in one of the python nix package options: 
```error: attribute ‘protobufBuild’ missing, at /Users/matt/proto3-suite/release.nix:16:12```

**Comment by intractable** [comment]
Yeah, I'm digging into this now.

**Comment by Gabriel439** [comment]
Try this:

```nix
python_protobuf3_0 = pkgs.pythonPackages.protobuf;
```

**Comment by themattchan** [comment]
Resolved by #50
